pkg/net/libp2p/key.go

```diff
@@ -49,11 +66,25 @@ func operatorPublicKeyToNetworkPublicKey(
 		return nil, fmt.Errorf("libp2p supports only secp256k1 operator keys")
 	}
 
-	return &libp2pcrypto.Secp256k1PublicKey{
-		Curve: DefaultCurve,
-		X:     operatorPublicKey.X,
-		Y:     operatorPublicKey.Y,
-	}, nil
+	operatorPublicKeyBytes := operator.MarshalCompressed(operatorPublicKey)
+
+	// Note that `libp2pcrypto.UnmarshalSecp256k1PublicKey` uses the secp256k1
+	// implementation provided by decred underneath
+	pubKey, err := libp2pcrypto.UnmarshalSecp256k1PublicKey(
+		operatorPublicKeyBytes,
+	)
+	if err != nil {
+		return nil, err
```

Comment:
```suggestion
		return nil, fmt.Errorf("cannot unmarshal operator public key as a libp2p secp256k1 public key: [%v]", err)
```

Reply:
Done in aebfadd057602ab9dcf5e0fc59e4e95b890d750a.